subsystems/viamserver/viamserver.go

```diff
@@ -59,9 +45,22 @@ type viamServer struct {
 	// for blocking start/stop/check ops while another is in progress
 	startStopMu sync.Mutex
 
+	// whether this viamserver instance handles needs restart checking itself; calculated
+	// and cached at startup; used by the manager to determine whether agent should handle
+	// needs restart checking on viamserver's behalf (this is the case for new viamserver
+	// versions)
+	doesNotHandleNeedsRestart bool
+
 	logger logging.Logger
 }
 
+type ctxKey int
+
+// CtxKeySendSIGQUITInsteadOfSIGTERM is used by manager to signal to the Stop method that
+// SIGQUIT should be sent instead of SIGTERM to get a goroutine dump and stop immediately
+// (when app has forcibly restarted agent).
+const CtxKeySendSIGQUITInsteadOfSIGTERM = ctxKey(iota)
```

Comment:
I don't _love_ this solution to tell `viam-server` to `Stop` using SIGQUIT instead of SIGTERM, but I didn't want to further change the `Subsystem` interface.